xla/service/elemental_ir_emitter_test.cc

```diff
@@ -77,6 +78,23 @@ class ElementalIrEmitterExecutionTestWithoutFastMinMax
   }
 };
 
+template <typename T>
+class ElementalIrEmitterExecutionTypedTest
+    : public ElementalIrEmitterExecutionTest {
+ protected:
+  const std::string& TypeName() {
+    return primitive_util::LowercasePrimitiveTypeName(
+        primitive_util::NativeToPrimitiveType<T>());
+  }
+};
+
+using FloatTypes =
+    ::testing::Types<bfloat16, tsl::float8_e4m3fn, tsl::float8_e4m3fnuz,
+                     tsl::float8_e4m3b11fnuz, tsl::float8_e5m2,
+                     tsl::float8_e5m2fnuz>;
+
+TYPED_TEST_SUITE(ElementalIrEmitterExecutionTypedTest, FloatTypes);
```

Comment:
Since we only use the type name, not the actual C++ type, this can be a value-parameterized test (with TEST_P) instead of a type-parameterized test (with TYPED_TEST)

Reply:
I initially attempted that approach, but I realized the `CompareFloat` test requires `TypeParam` (used in `LiteralUtil::CreateR1<TypeParam>`) to execute `RunTest` with a Span of Literals of the specified type. Therefore, I refactored the original implementation using `TYPED_TEST_SUITE`.